src/libs/import.c

```diff
@@ -1665,20 +1665,21 @@ static void _browse_basedir_clicked(GtkWidget *widget, GtkEntry *basedir)
   {
     topwindow = dt_ui_main_window(darktable.gui->ui);
   }
-  GtkWidget *filechooser = gtk_file_chooser_dialog_new(
-      _("select directory"), GTK_WINDOW(topwindow), GTK_FILE_CHOOSER_ACTION_SELECT_FOLDER, _("_cancel"),
-      GTK_RESPONSE_CANCEL, _("_open"), GTK_RESPONSE_ACCEPT, (char *)NULL);
+
+  GtkFileChooserNative *filechooser = gtk_file_chooser_native_new(
+      _("select directory"), GTK_WINDOW(topwindow), GTK_FILE_CHOOSER_ACTION_SELECT_FOLDER, _("_open"), _("_cancel"));
+
+/* is this still needed ?
 #ifdef GDK_WINDOWING_QUARTZ
   dt_osx_disallow_fullscreen(filechooser);
 #endif
-
-  gtk_file_chooser_set_select_multiple(GTK_FILE_CHOOSER(filechooser), FALSE);
+*/
```

Comment:
select single for file chooser for basedir (or any dir) is needed. as for disallowing fulscreen for native chooser? that is question to @parafin :)

Reply:
> select single for file chooser for basedir (or any dir) is needed

Sure single it is not the default for dirs? There is no way I can select multiple items in the chooser

Reply:
I think you're right. delete it then instead of commenting out ;)

Reply:
dt_osx_disallow_fullscreen is NOP since GTK 3.24.14, so it's difficult to test this. But I think these calls should be removed, since this function can only be called on GtkWidget and GtkFileChooserNative is not a GtkWidget.